File: 프로그래머스/2/12902. 3 x n 타일링/3 x n 타일링.py
def solution(n):
    if n % 2 != 0: return 0

    dp = [0] * (n + 1)
    dp[0] = 1
    total = 0

    for i in range(2, n + 1, 2):
        dp[i] = (dp[i-2] * 3 + total * 2) % 1000000007
        total += dp[i-2]

    return dp[n] % 1000000007

File: 프로그래머스/2/12902. 3 x n 타일링/3 x n 타일링.js
function solution(n) {
    if (n % 2 !== 0) return 0;

    let i = 0;
    let dp = [1];
    let sum = 0;
    for(i = 2; i <= n; i += 2) {
        dp[i] = (dp[i-2] * 3 + sum * 2) % 1000000007;
        sum += dp[i-2];
    }

    return dp[n] % 1000000007;
}